src/io/flutter/perf/HeapMonitor.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.perf;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.intellij.openapi.diagnostic.Logger;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.IsolatesInfo;
+import io.flutter.run.FlutterDebugProcess;
+import org.dartlang.vm.service.VmService;
+import org.dartlang.vm.service.consumer.GetIsolateConsumer;
+import org.dartlang.vm.service.element.Isolate;
+import org.dartlang.vm.service.element.Obj;
+import org.dartlang.vm.service.element.RPCError;
+import org.dartlang.vm.service.element.Sentinel;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+// TODO(pq): handle GC events
```

Comment:
We should move to a time based graph (time as the x axis instead of sample count). That'll let us do things like incorporate GC event info, without distorting the graph.

Reply:
Ah.  I was probably oversimplifying, thinking that we could infer time from the count (since we're polling at a steady rate).  Let's discuss in person.  (I'm open to whatever!)